tests/console/cryptsetup.pm

```diff
@@ -0,0 +1,55 @@
+# SUSE's openQA tests
+#
+# Copyright © 2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Case 1528909  - FIPS: cryptsetup
+
+use base "consoletest";
+use strict;
+use testapi;
+
+sub run() {
+
+    my $slef        = shift;
+    my $cryptpasswd = 'susetesting';
+    select_console "root-console";
+
+    # create a random volume.
+    assert_script_run("dd if=/dev/urandom of=/test.dm bs=1k count=51200");
+    wait_still_screen;
+
+    # use cryptsetup to luksFormat and luksOpen volume
+    assert_script_run("echo -e $cryptpasswd | cryptsetup -q luksFormat /test.dm");
+    wait_still_screen;
+    assert_script_run("echo -e $cryptpasswd | cryptsetup -q luksOpen /test.dm dmtest");
+    wait_still_screen;
+
+    # format the dmtest and mount it with /test
+    assert_script_run("mkfs.ext4 /dev/mapper/dmtest");
+    wait_still_screen;
+    assert_script_run("mkdir /test;mount /dev/mapper/dmtest /test");
+    wait_still_screen;
+
+    # build some directory and files in the mount point ,try to catch file and delete directory and file
+    assert_script_run("cd /test");
+    assert_script_run("for x in `seq 100`; do mkdir d\$x; date > d\$x/f\$x.log; done");
```

Comment:
As you don't want perl to evaluate the variables I recommend to use single ticks instead, e.g.

```perl
assert_script_run('for x in `seq 100`; do mkdir d$x; date > d$x/f$x.log; done');
```